python/lsst/daf/butler/script/certifyCalibrations.py

```diff
@@ -0,0 +1,72 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import astropy.time
+
+from .. import Butler, CollectionType, Timespan
+
+
+def certifyCalibrations(repo, input_collection, output_collection, dataset_type_name,
+                        begin_date, end_date, search_all_inputs):
+    """Certify a set of calibrations with a validity range.
+
+    Parameters
+    ----------
+    repo : `str`
+        URI to the location of the repo or URI to a config file describing the
+        repo and its location.
+    inputCollection : `str`
+       Data collection to pull calibrations from.  Usually an existing
+        `~CollectionType.RUN` or `~CollectionType.CHAINED` collection, and may
+        _not_ be a `~CollectionType.CALIBRATION` collection or a nonexistent
+        collection.
+    outputCollection : `str`
+        Data collection to store final calibrations.  If it already exists, it
+        must be a `~CollectionType.CALIBRATION` collection.  If not, a new
+        `~CollectionType.CALIBRATION` collection with this name will be
+        registered.
+    datasetTypeName : `str`
+        Name of the dataset type to certify.
+    begin_date : `str`, optional
+        Date this calibration should start being used.
+    end_date : `str`, optional
+        Date this calibration should stop being used.
+    search_all_inputs : `bool`, optional
+        Search all children of the inputCollection if it is a CHAINED
+        collection, instead of just the most recent one.
+    """
+    butler = Butler(repo, writeable=True)
+    registry = butler.registry
+    timespan = Timespan(
+        begin=astropy.time.Time(begin_date) if begin_date is not None else None,
```

Comment:
Might be more explicit to add `scale="tai"` here and below so we are completely certain these are parsed as TAI.

@TallJimbo should we be supporting the same syntax we support for queries? ie https://pipelines.lsst.io/v/daily/modules/lsst.daf.butler/queries.html#time-literals ? (the leading T is not needed of course -- I mean the `mjd/` and `/tai` stuff -- we have code for converting that string to an astropy Time.

Maybe we support ISO string now and later on can expand if asked (because an ISO string is compatible with how the T-string works).